pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1069,6 +1073,10 @@ IdealState ensureAllPartitionsConsuming(TableConfig tableConfig, PartitionLevelS
     //       and restart consumption from the same offset (if possible) or a newer offset (if realtime stream does
     //       not have the same offset).
     //       In latter case, report data loss.
+    // 4. There are no segments in the table's idealstate in CONSUMING state,
```

Comment:
The new scenario is very similar to 1, so we can keep them together:
We first check if the `latestSegmentZKMetadata` is `DONE`, then check if `instanceStateMap` contains `CONSUMING` segment.

Reply:
Ack